esx_service/vmdk_ops.py

```diff
@@ -1318,10 +1325,11 @@ def disk_detach(vmdk_path, vm):
     if not device:
        # Could happen if the disk attached to a different VM - attach fails
        # and docker will insist to sending "unmount/detach" which also fails.
-       msg = "*** Detach failed: disk={0} not found. VM={1}".format(
-       vmdk_path, vm.config.uuid)
-       logging.warning(msg)
-       return err(msg)
+       # Or Plugin retrying operation due to socket errors #1076
+       # Return success since disk is anyway not attached
+       logging.warning("*** Detach disk={0} not found. VM={1}".format(
+                       vmdk_path, vm.config.uuid))
+       return None
```

Comment:
Test is also failing that the volume is in use when remove is issued without a detach. That path is not addressable as we can;t auto detach the volume. Or can we?